### PR TITLE
Ret 4281 new

### PR DESCRIPTION
**Pull request by btensay**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/RET-4281

### Change description ###
doc upload event added

### Checklist

- [x ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [x ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DocumentManagementService.java [hunk 1]
```diff
@@ -253,6 +260,34 @@ private ResponseEntity<DocumentDetails> getDocumentDetails(String authToken, UUI
...
+                    setSecondLevelDocumentFromType(uploadedDocType, uploadedDocType.getDocumentType());
+                    DocumentTypeItem docTypeItem = createDocumentTypeItemFromTopLevel(
+                            uploadedDocType.getUploadedDocument(),
+                            uploadedDocType.getTopLevelDocuments(),
+                            uploadedDocType.getDocumentType(),
+                            String.format("%s : %s", uploadedDocType.getShortDescription(),
+                                    uploadedDocType.getTopLevelDocuments()));
+                    docTypeItem.getValue().setDateOfCorrespondence(uploadDoc.getValue().getDateOfCorrespondence());
+                    caseData.getDocumentCollection().add(docTypeItem);
+                    setDocumentNumbers(caseData);
```
Comment (LahiruOshara) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DocumentManagementService.java:287]: setDocumentNumbers method iterate through the caseData document collection. should move the method call out of the loop I think

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/AddDocumentControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    void testAboutToSubmitWithInvalidToken() throws Exception {
+        when(verifyTokenService.verifyTokenSignature("invalidToken")).thenReturn(false);
+        mockMvc.perform(post(ABOUT_TO_SUBMIT_URL)
+                        .contentType(APPLICATION_JSON)
+                        .header(HttpHeaders.AUTHORIZATION, AUTH_TOKEN)
+                        .content(jsonMapper.toJson(ccdRequest)))
+                .andExpect(status().isForbidden());
+        verify(documentManagementService, never()).addUploadedDocsToCaseDocCollection(any());
+    }
+}
```
Comment (LahiruOshara) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/AddDocumentControllerTest.java:90]:
```suggestion
}

```

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DocumentManagementService.java (hunk 1 above)
Comment (btensay) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/DocumentManagementService.java]: thanks. Done

**Review by hsjhita — APPROVED** [review]

**Review by LahiruOshara — APPROVED** [review]

**Review by hsjhita — DISMISSED** [review]

File: src/main/resources/application-dev.yaml [hunk 3]
```diff
@@ -27,4 +27,4 @@ ccd:
   data-store-api-url: http://localhost:4452
 
 case_document_am:
-  url: http://localhost:4455
+  url: http://localhost:4455
```
Comment (LahiruOshara) [inline src/main/resources/application-dev.yaml:30]:
```suggestion
  url: http://localhost:4455
  
```

**Pull request merged**